modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/ClusterProcessor.java

```diff
@@ -397,30 +390,7 @@ public void initDiagnosticListeners() throws IgniteCheckedException {
                             return;
                         }
 
-                        byte[] diagRes;
-
-                        try {
-                            CompoundInfo info = msg0.unmarshal(marsh);
-
-                            diagRes = marsh.marshal(info.diagnosticInfo(ctx));
-                        }
-                        catch (Exception e) {
-                            U.error(diagnosticLog, "Failed to run diagnostic closure: " + e, e);
-
-                            try {
-                                IgniteDiagnosticInfo errInfo =
-                                    new IgniteDiagnosticInfo("Failed to run diagnostic closure: " + e);
-
-                                diagRes = marsh.marshal(errInfo);
-                            }
-                            catch (Exception e0) {
-                                U.error(diagnosticLog, "Failed to marshal diagnostic closure result: " + e, e);
-
-                                diagRes = null;
-                            }
-                        }
-
-                        IgniteDiagnosticMessage res = IgniteDiagnosticMessage.createResponse(diagRes, msg0.futureId());
+                        IgniteDiagnosticMessage res = new IgniteDiagnosticMessage(msg0.diagnosticInfo(ctx), msg0.futureId());
```

Comment:
I suggest not to use one message for request and for response. Suggensted approach (if possible):

1. Rename `IgniteDiagnosticMessage` to `IgniteDiagnosticRequest`. It should keep diagnosic infos.
2. Add `String` response holder, eg. `IgniteDiagnosticRequest`.